pkg/controller/atlasdatabaseuser/databaseuser.go

```diff
@@ -218,52 +208,59 @@ func filterScopeDeployments(user akov2.AtlasDatabaseUser, allDeploymentsInProjec
+	matches, err := userMatchesSpec(log, &atlasUser.AtlasDatabaseUserSpec, &operatorUser.Spec)
 	if err != nil {
 		return false, err
 	}
 	if !matches {
 		return true, nil
 	}
 	// We need to check if the password has changed since the last time
-	passwordsChanged := operatorDBUser.Status.PasswordVersion != currentPasswordResourceVersion
+	passwordsChanged := operatorUser.Status.PasswordVersion != currentPasswordResourceVersion
 	if passwordsChanged {
 		log.Debug("Database User password has changed - making the request to Atlas")
 	}
 	return passwordsChanged, nil
 }
 
-// TODO move to a separate utils (reuse from deployments)
-func userMatchesSpec(log *zap.SugaredLogger, atlasSpec *mongodbatlas.DatabaseUser, operatorSpec akov2.AtlasDatabaseUserSpec) (bool, error) {
-	userMerged := mongodbatlas.DatabaseUser{}
-	if err := compat.JSONCopy(&userMerged, atlasSpec); err != nil {
+func userMatchesSpec(log *zap.SugaredLogger, atlasUsername, operatorUser *akov2.AtlasDatabaseUserSpec) (bool, error) {
+	operatorCopy, err := dbuser.Normalize(operatorUser.DeepCopy())
+	if err != nil {
 		return false, err
 	}
 
-	if err := compat.JSONCopy(&userMerged, operatorSpec); err != nil {
-		return false, err
+	if atlasUsername.Username != operatorCopy.Username {
+		log.Infof("Usernames differs from spec: %q <> %q\n", atlasUsername.Username, operatorCopy.Username)
+		return false, nil
 	}
-
-	// performing some normalization of dates
-	if atlasSpec.DeleteAfterDate != "" {
-		atlasDeleteDate, err := timeutil.ParseISO8601(atlasSpec.DeleteAfterDate)
-		if err != nil {
-			return false, err
-		}
-		atlasSpec.DeleteAfterDate = timeutil.FormatISO8601(atlasDeleteDate)
+	if atlasUsername.DatabaseName != operatorCopy.DatabaseName {
+		log.Infof("DatabaseName differs from spec: %q <> %q\n", atlasUsername.DatabaseName, operatorCopy.DatabaseName)
+		return false, nil
 	}
-	if operatorSpec.DeleteAfterDate != "" {
-		operatorDeleteDate, err := timeutil.ParseISO8601(operatorSpec.DeleteAfterDate)
-		if err != nil {
-			return false, err
-		}
-		userMerged.DeleteAfterDate = timeutil.FormatISO8601(operatorDeleteDate)
+	if atlasUsername.DeleteAfterDate != operatorCopy.DeleteAfterDate {
+		log.Infof("DeleteAfterDate differs from spec: %q <> %q\n", atlasUsername.DeleteAfterDate, operatorCopy.DeleteAfterDate)
+		return false, nil
 	}
-	d := cmp.Diff(*atlasSpec, userMerged, cmpopts.EquateEmpty())
-	if d != "" {
-		log.Debugf("Users differs from spec: %s", d)
+	if atlasUsername.OIDCAuthType != operatorCopy.OIDCAuthType {
+		log.Infof("OIDCAuthType differs from spec: %q <> %q\n", atlasUsername.OIDCAuthType, operatorCopy.OIDCAuthType)
+		return false, nil
 	}
-
-	return d == "", nil
+	if atlasUsername.AWSIAMType != operatorCopy.AWSIAMType {
+		log.Infof("AWSIAMType differs from spec: %q <> %q\n", atlasUsername.AWSIAMType, operatorCopy.AWSIAMType)
```

Comment:
nit: instead of logging here, return a `diff string` and the caller can decide whether this needs logging, event emission, increasing a "diff" counter, etc. etc.

Reply:
will fix

Reply:
done